SYS-1554: fix logging in bookplate update scripts

ztucker4 opened the pull request:
Implements [SYS-1554](https://uclalibrary.atlassian.net/browse/SYS-1554)

Several logging-related changes to `add_bib_ebookplates.py` and `update_bookplates_one_time.py`:
* Log level is now set with the command line argument `--log-level`, with possible values `"DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"`. `INFO` is default.
* Logs from `urllib3` with lower priority than `WARNING` are now suppressed.
* All print statements have been changed to logs at the `INFO` level.
* Report index is now logged every 1% of records for easier rerunning in production.
* The optional `--start-index` argument is now supported in `add_bib_ebookplates.py`.

[SYS-1554]: https://uclalibrary.atlassian.net/browse/SYS-1554?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

akohler commented on add_bib_ebookplates.py:
```diff
@@ -210,22 +224,19 @@ def main():
         )
-        # Take 5%, round down, add 1 to avoid 0 when length < 20
-        progress_interval = (len(report_with_ebookplates) // 20) + 1
+        # Take 1%, round down, add 1 to avoid 0 when length < 20
```
Update comment to match code: `20` should now be `100`.

akohler commented on update_bookplates_one_time.py:
```diff
@@ -171,21 +178,18 @@ def main():
-        # Take 5%, round down, add 1 to avoid 0 when length < 20
-        progress_interval = (len(report) // 20) + 1
+        # every 1% of records, log progress
+        # Take 1%, round down, add 1 to avoid 0 when length < 20
```
Update comment to match code: `20` should now be `100`

akohler reviewed (changes requested):
@ztucker4 Two very minor changes, the same in both scripts: When values are used in both code and comments, updates have to be kept in sync.  Or reword comment to avoid specifics, maybe "avoid 0 when report length is small" or similar.

One other possible change, up to you: The 0-based `report_index` can lead to confusing (to me...) log output.  I ran this in `test` mode, with this output:
```
INFO:root:Beginning processing 2 bib e-bookplates
INFO:root:Processed 0 bibs.
INFO:root:Skipping MMS ID 9990572683606533. No 966 updates needed.
INFO:root:Processed 1 bibs.
INFO:root:Finished adding ebookplates.
INFO:root:1 bibs updated.
INFO:root:1 bibs skipped with no 966 updates needed.
INFO:root:0 bibs skipped due to errors.
```
It's somewhat confusing that the number of bibs processed starts at 0, especially since that message is logged **after** at least one record is processed.  Maybe change `report_index` initialization to start at 1?  Or consider whether this is useful at all: if not running in `DEBUG` mode, if the program crashes, how do we know where to restart it?  In the default `INFO` logging mode, I think the program only logs errors and (unexpected, normally) skips.  It's probably better to log every row by default, to make it easy to determine what the last processed MMS ID is.

ztucker4 commented:
Hi @akohler, this is ready for another look. I've fixed the outdated comments. For the report indexing issue, I've changed `update_bookplates_one_time.py` to increment the index before the percent calculation is done and the item is potentially logged. Now logs should always start at (at least) 1.
I've also added the last processed MMS ID to the logs written every 1% in both scripts. Since this is at the default `INFO` level, this will allow us to start reruns at most 1% back from the actual failed record. The script skips already-updated bibs, so rerunning MMS IDs shouldn't cause any problems.
